### PR TITLE
initial version

### PR DESCRIPTION
**Pull request by trapacska**

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    envs:
+    - BITRISE_STEP_GIT_CLONE_URL: https://github.com/godrei/steps-github-release.git
+    - BITRISE_STEP_ID: github-release
+    - BITRISE_STEP_VERSION: $BITRISE_GIT_TAG
+    - BITRISE_GITHUB_API_TOKEN: $BITRISE_GITHUB_API_TOKEN
+    - MY_STEPLIB_REPO_FORK_GIT_URL: $MY_STEPLIB_REPO_FORK_GIT_URL
+    before_run:
+    - test
+    steps:
+    - git::https://github.com/godrei/steps-generate-changelog@master:
```
Comment (trapacska) [inline bitrise.yml]: replace this with the actual step

File: bitrise.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        inputs:
+        - new_version: $BITRISE_STEP_VERSION
+    - git::https://github.com/godrei/steps-github-release@master:
```
Comment (trapacska) [inline bitrise.yml]: replace this with the actual step

File: main.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+// formats:
+// https://hostname/owner/repository.git
+// git@hostname:owner/repository.git
+// ssh://git@hostname:port/owner/repository.git
+func parseRepo(url string) (host string, owner string, name string) {
```
Comment (trapacska) [inline main.go]: `func parseRepo(url string) (host string, owner string,`**`name string)`** is not used

File: main.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+func main() {
+	apiToken := os.Getenv("api_token")
+	repositoryURL := os.Getenv("repository_url")
+	tag := os.Getenv("tag")
+	commit := os.Getenv("commit")
+	name := os.Getenv("name")
+	body := os.Getenv("body")
+	draft := os.Getenv("draft")
```
Comment (trapacska) [inline main.go]: use stepconf

File: main_test.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseRepo(t *testing.T) {
+	t.Log("Parses: https://hostname/owner/repository.git")
+	{
+		host, owner, name := parseRepo("https://github.com/godrei/steps-github-release.git")
```
Comment (trapacska) [inline main_test.go]: change to another org

File: step.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
+title: Github Release
+summary: Create Github Release
+description: Create Github Release
+
+website: https://github.com/godrei/steps-github-release
```
Comment (trapacska) [inline step.yml]: change them

File: step.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  opts:
+    title: Tag
+    summary: The name of the tag.
+    is_required: true
+- commit: $BITRISE_GIT_COMMIT
+  opts:
+    title: Commit
+    summary: The commit of the tag
+    description: |-
+      Specifies the commitish value that determines where the Git tag is created from.
```
Comment (trapacska) [inline step.yml]: commitish?

File: step.yml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    summary: The name of the release.
+    is_required: true
+- body:
+  opts:
+    title: Release body
+    summary: The body of the release.
+    is_required: true
+- draft: yes
+  opts:
+    title: Release body
```
Comment (trapacska) [inline step.yml]: it is draft

File: LICENSE [hunk 9]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (BirmacherAkos) [inline LICENSE]: Ejnye bejnye :D

File: main.go [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+	log.Errorf(format, args...)
+	os.Exit(1)
+}
+
+// Config ...
+type Config struct {
+	APIToken      stepconf.Secret `env:"api_token,required"`
+	Username      stepconf.Secret `env:"username,required"`
+	RepositoryURL string          `env:"repository_url,required"`
+	Tag           string          `env:"tag,required"`
```
Comment (BirmacherAkos) [inline main.go]: Are you sure, that every input is required?

File: main.go [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+	Commit        string          `env:"commit,required"`
+	Name          string          `env:"name,required"`
+	Body          string          `env:"body,required"`
+	Draft         string          `env:"draft,opt[yes,no]"`
+}
+
+// RoundTrip ...
+func (c Config) RoundTrip(req *http.Request) (*http.Response, error) {
+	userPassPair := []byte(fmt.Sprintf("%s:%s", string(c.Username), string(c.APIToken)))
+	req.Header.Set("Authorization", "Basic "+base64.StdEncoding.EncodeToString(userPassPair))
```
Comment (BirmacherAkos) [inline main.go]: https://golang.org/pkg/net/http/#Request.BasicAuth

**Pull request merged**